interfaces/cython/cantera/soln2ck.py

```diff
@@ -0,0 +1,362 @@
+        line_coeffs = ''
+        lines = [[1, 2, 3, 4, 5], [6, 7, 8, 9, 10], [11, 12, 13, 14]]
+        line_index = lines[row-2]
+        for ix, c in enumerate(nasa_coeffs):
+            if ix in line_index:
+                if c >= 0:
+                    line_coeffs += ' '
+                line_coeffs += str('{:.8e}'.format(c))
+        return line_coeffs
+
+    def build_species_string():
+        """
+        formats species string for writing
+        """
+        species_list_string = ''
+        line = 1
+        for sp_index, sp_string in enumerate(trimmed_solution.species_names):
+            sp = ' '
+            #get length of string next species is added
+            length_new = len(sp_string)
+            length_string = len(species_list_string)
+            total = length_new + length_string + 3
+            #if string will go over width, wrap to new line
+            if total >= 70*line:
+                species_list_string += '\n'
+                line += 1
+            species_list_string += sp_string + ((16-len(sp_string))*sp)
+        return species_list_string
+
+    #Write title block to file
+    section_break('Chemkin File converted from Solution Object by pyMARS')
+
+    #Write phase definition to file
+    element_names = eliminate(str(trimmed_solution.element_names),
+                                        ['[', ']', '\'', ','])
+    element_string = Template(
+                            'ELEMENTS\n' +
+                            '$element_names\n' +
+                            'END\n')
+    f.write(element_string.substitute(element_names=element_names))
+    species_names = build_species_string()
+    species_string = Template(
+                            'SPECIES\n' +
+                            '$species_names\n'+
+                            'END\n')
+    f.write(species_string.substitute(species_names=species_names))
+
+    #Write species to file
+    section_break('Species data')
+    f.write('THERMO ALL' + '\n' +
+            '   300.000  1000.000  5000.000' +'\n')
+    phase_unknown_list = []
+
+    #write data for each species in the Solution object
+    for sp_index in xrange(len(trimmed_solution.species_names)):
+        d = 3.33564e-30 #1 debye = d coulomb-meters
+        species = trimmed_solution.species(sp_index)
+        name = str(trimmed_solution.species(sp_index).name)
+        nasa_coeffs = trimmed_solution.species(sp_index).thermo.coeffs
+        #Species attributes from trimmed solution object
+        t_low = '{0:.3f}'.format(species.thermo.min_temp)
+        t_max = '{0:.3f}'.format(species.thermo.max_temp)
+        t_mid = '{0:.3f}'.format(species.thermo.coeffs[0])
+        temp_range = str(t_low) + '  ' + str(t_max) + '  ' + t_mid
+        species_comp = ''
+        for atom in species.composition:
+            species_comp += '{:<4}'.format(atom)
+            species_comp += str(int(species.composition[atom]))
+        if type(species.transport).__name__ == 'GasTransportData':
```

Comment:
This actually isn't totally correct, we do support Chemkin surface kinetics, but I think for a first version, only supporting gas phase is fine.